### PR TITLE
Homework week 3 Osama Bazbooz

### PR DESCRIPTION
**Pull request by obazbooz**

### CODE REVIEW

File: 1-JavaScript/Week3/homework/ex1-giveCompliment.js [hunk 1]
```diff
@@ -18,12 +18,25 @@ Full description at: https://github.com/HackYourFuture/Homework/tree/main/1-Java
    Use `console.log` each time to display the return value of the
    `giveCompliment` function to the console.
 -----------------------------------------------------------------------------*/
-function giveCompliment(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+function giveCompliment(name) {
```
[the remark this answers is not in the document]
Comment (obazbooz) [reply 1-JavaScript/Week3/homework/ex1-giveCompliment.js:21]: Thanks a lot @killoncio  :)

File: 1-JavaScript/Week3/homework/ex2-dogYears.js [hunk 2]
```diff
@@ -16,8 +16,9 @@ calculate it!
    ages.
 -----------------------------------------------------------------------------*/
 
-function calculateDogAge(/* parameters go here */) {
-  // TODO complete this function
+function calculateDogAge(number) {
```
[the remark this answers is not in the document]
Comment (obazbooz) [reply 1-JavaScript/Week3/homework/ex2-dogYears.js:19]: 👍 Thanks

File: 1-JavaScript/Week3/homework/ex3-tellFortune.js [hunk 3]
```diff
@@ -30,30 +30,28 @@ Note: The DRY principle is put into practice here: instead of repeating the code
...
-  // TODO complete this function
+function selectRandomly(fortuneArray) {
+  const randomElement = Math.floor(Math.random() * fortuneArray.length);
+  const random = fortuneArray[randomElement];
+  return random;
 }
 
-function tellFortune(/* add parameter(s) here */) {
-  // TODO complete this function
+function tellFortune(numKids, partnerNames, locations, jobTitles) {
```
[the remark this answers is not in the document]
Comment (obazbooz) [reply 1-JavaScript/Week3/homework/ex3-tellFortune.js:49]: :D

File: 1-JavaScript/Week3/homework/ex4-shoppingCart.js [hunk 4]
```diff
@@ -23,8 +23,18 @@ you have more than 3 items in your shopping cart the first item gets taken out.
...
 // ! Function to be tested
-function addToShoppingCart(/* parameters go here */) {
-  // TODO complete this function
+function addToShoppingCart(groceryItem) {
+  if (groceryItem === undefined) {
+    return `You bought ${shoppingCart[0]}, ${shoppingCart[1]}!`;
+  } else {
+    if (shoppingCart.length === 3) {
+      shoppingCart.shift();
+      shoppingCart.push(groceryItem);
```
[the remark this answers is not in the document]
Comment (obazbooz) [reply 1-JavaScript/Week3/homework/ex4-shoppingCart.js]: You are right.. I have corrected it and put it out of the if/else condition 👍

File: 1-JavaScript/Week3/homework/ex5-shoppingCartPure.js [hunk 5]
```diff
@@ -16,8 +16,11 @@ it pure. Do the following:
 5. Confirm that you function passes the provided unit tests.
 ------------------------------------------------------------------------------*/
 // ! Function under test
-function addToShoppingCart(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+function addToShoppingCart(shoppingCart, groceryItem) {
+  if (shoppingCart.length > 2) {
+    shoppingCart.pop();
```
[the remark this answers is not in the document]
Comment (obazbooz) [reply 1-JavaScript/Week3/homework/ex5-shoppingCartPure.js]: I have created new local array and assigned the original one to for the modification proposes.
Thanks a lot of your feedback @killoncio

File: 1-JavaScript/Week3/homework/ex6-totalCost.js [hunk 6]
```diff
@@ -21,21 +21,29 @@ instead!
...
+  chips: 2,
+  cola: 1.7,
+  marshmello: 0.7,
+  tea: 4,
+  pistachio: 3,
 };
 
-function calculateTotalPrice(/* TODO parameter(s) go here */) {
-  // TODO replace this comment with your code
+function calculateTotalPrice(receipt) {
```
[the remark this answers is not in the document]
Comment (obazbooz) [reply 1-JavaScript/Week3/homework/ex6-totalCost.js:31]: :)

File: 1-JavaScript/Week3/homework/ex4-shoppingCart.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (killoncio) [reply 1-JavaScript/Week3/homework/ex4-shoppingCart.js]: Yes, now it's correct! :) 

Regarding `shoppingCart.length === 3`, normally it's a better idea to use `shoppingCart.length > 2`, to make sure you cover all cases. In this exercise seems quite straightforward that there is no way the list could go from 2 to 4 skipping the 3, but there could be more complex functions where that could happen.

File: 1-JavaScript/Week3/homework/ex5-shoppingCartPure.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (killoncio) [reply 1-JavaScript/Week3/homework/ex5-shoppingCartPure.js]: Yes, that's correct!

You are returning though both in the `if` and in the `else` , has you were doing in previous exercise at the beginning. Also here is better to always return, meaning outside the if/else block.

**Pull request closed without merging**